custom/covid/management/commands/update_cases.py

```diff
@@ -0,0 +1,44 @@
+from django.core.management.base import BaseCommand
+
+from corehq.apps.linked_domain.dbaccessors import get_linked_domains
+from corehq.apps.users.util import SYSTEM_USER_ID
+from corehq.apps.users.util import username_to_user_id
+from corehq.form_processor.interfaces.dbaccessors import CaseAccessors
+
+
+class CaseUpdateCommand(BaseCommand):
+    """
+        Base class for updating cases of a specific case_type.
+        Override all methods that raise NotImplementedError.
+    """
+
+    def case_block(self):
+        raise NotImplementedError()
+
+    def update_cases(self, domain, case_type, user_id):
+        raise NotImplementedError()
+
+    def find_case_ids_by_type(self, domain, case_type):
+        accessor = CaseAccessors(domain)
+        case_ids = accessor.get_case_ids_in_domain(case_type)
+        print(f"Found {len(case_ids)} {case_type} cases in {domain}")
+        return case_ids
+
+    def add_arguments(self, parser):
+        parser.add_argument('domain')
+        parser.add_argument('case_type')
+        parser.add_argument('username')
```

Comment:
It would be sweet if this were optional rather than falling back to the system user if the username is invalid. Then the intent of the user is clear. You could leave it off if you did not want it, rather than using `areallyfakeuserthatdefinitelydoesnotexist` as I did today, but also have stronger catching so that if a user is included, we make sure its valid and a misspelled username won't cause all the updates to be attributed to the system user.

Reply:
[e344c90](https://github.com/dimagi/commcare-hq/pull/28996/commits/e344c90284a94ebf0c59e6543ab42cb8aae5d000) with added test